src/test/java/BaseTest.java

```diff
@@ -1,11 +1,145 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
-import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+import java.util.*;
+
+import java.time.Duration;
 
 public class BaseTest {
 
+    public static WebDriver driver = null;
+    public static WebDriverWait wait = null;
+    public static String url = "https://qa.koel.app/";
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters ({"BaseURL"})
+    public void launchBrowser(String BaseURL) throws InterruptedException{
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        url = BaseURL;
+        driver.get(url);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+
+    }
+    @AfterMethod
+    public void logout() {
+
+        driver.quit();
+    }
+    @DataProvider (name="CorrectLoginProviders")
+    public static Object[][] getDataFromDataProviders(){
+        return new Object[][] {
+                {"matt.pierce@testpro.io", "te$t$tudent"}
+        };
+    }
+
+    public static void openLoginUrl() {
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+    }
+
+    protected static void enterEmail(String email) throws InterruptedException{
+        WebElement emailInput = driver.findElement(By.cssSelector("[type='email']"));
+        emailInput.click();
+        emailInput.clear();
+        //inputs valid email address into email input field
+        emailInput.sendKeys(email);
+        Thread.sleep(5000);
```

Comment:
Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.